common/txmgr/inmemory_store.go

```diff
@@ -334,7 +336,53 @@ func (ms *inMemoryStore[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) FindT
 	return nil, nil
 }
 func (ms *inMemoryStore[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) MarkAllConfirmedMissingReceipt(ctx context.Context, chainID CHAIN_ID) error {
```

Comment:
As part of EVM refactor, in future, this function should go away into chain-specific code. The TxStore shouldn't need to know nonce specific logic which it is doing here currently.
Not for now though.